### PR TITLE
Documentation for texels per unit on BakedLightmap.

### PR DESCRIPTION
**Pull request by NathanWarden**

### CODE REVIEW

File: doc/classes/BakedLightmap.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
 			Grid subdivision size for lightmapper calculation. The default value will work for most cases. Increase for better lighting on small details or if your scene is very large.
 		</member>
 		<member name="bake_default_texels_per_unit" type="float" setter="set_bake_default_texels_per_unit" getter="get_bake_default_texels_per_unit" default="20.0">
+			If a [code]lightmap_size_hint[/code] isn't specified on a [code]Mesh[/code] then the lightmap baker will dynamically set the lightmap size using this value. This value is measured in texels per world unit. The maximum lightmap texture size is clamped at 4096x4096.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]:
```suggestion
			If a [code]lightmap_size_hint[/code] isn't specified on a [code]Mesh[/code], the lightmap baker will dynamically set the lightmap size using this value. This value is measured in texels per world unit. The maximum lightmap texture size is 4096×4096.
```

**Review by Calinou — CHANGES_REQUESTED** [review]
Looks good to me, please take a look at the suggestion I made :slightly_smiling_face:

File: doc/classes/BakedLightmap.xml (hunk 1 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: I like that change so I'll change it to that. I'm going to go ahead and fill out the rest of the docs for the BakedLightmap node as well while I'm at it.

**Comment by NathanWarden** [comment]
Okay, documented the rest of the items as well. :)

File: doc/classes/BakedLightmap.xml [hunk 2]
```diff
@@ -18,12 +18,14 @@
 			<argument index="1" name="create_visual_debug" type="bool" default="false">
 			</argument>
 			<description>
+				Bakes the lightmaps within the currently editted scene.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]:
```suggestion
				Bakes the lightmaps within the currently-edited scene.
```

File: doc/classes/BakedLightmap.xml [hunk 3]
```diff
@@ -18,12 +18,14 @@
...
 			</description>
 		</method>
 		<method name="debug_bake">
 			<return type="void">
 			</return>
 			<description>
+				Executes a dry run bake of lightmaps within the currently editted scene.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]:
```suggestion
				Executes a dry run bake of lightmaps within the currently-edited scene.
```

File: doc/classes/BakedLightmap.xml [hunk 4]
```diff
@@ -32,8 +34,10 @@
 			Grid subdivision size for lightmapper calculation. The default value will work for most cases. Increase for better lighting on small details or if your scene is very large.
 		</member>
 		<member name="bake_default_texels_per_unit" type="float" setter="set_bake_default_texels_per_unit" getter="get_bake_default_texels_per_unit" default="20.0">
+			If a [code]lightmap_size_hint[/code] isn't specified on a [code]Mesh[/code], the lightmap baker will dynamically set the lightmap size using this value. This value is measured in texels per world unit. The maximum lightmap texture size is 4096x4096.
 		</member>
 		<member name="bake_energy" type="float" setter="set_energy" getter="get_energy" default="1.0">
+			Multiplies the light intensity of light sources by this value. For instance, if the value is set to 2 the light will start twice as bright. If it is 0.5 it will be half as bright.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]:
```suggestion
			Multiplies the light sources' intensity by this value. For instance, if the value is set to 2, lights will be twice as bright. If the value is set to 0.5, lights will be half as bright.
```

File: doc/classes/BakedLightmap.xml [hunk 5]
```diff
@@ -45,6 +49,7 @@
 			Lightmapping mode. See [enum BakeMode].
 		</member>
 		<member name="bake_propagation" type="float" setter="set_propagation" getter="get_propagation" default="1.0">
+			Defines how far the light will travel before it is no longer effective. The higher the number the farther the light will travel. For instance, if the value is set to 2 the light will go twice as far. If it is 0.5 it will go half as far.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]:
```suggestion
			Defines how far the light will travel before it is no longer effective. The higher the number, the farther the light will travel. For instance, if the value is set to 2, the light will go twice as far. If the value is set to 0.5, the light will only go half as far.
```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: doc/classes/BakedLightmap.xml (hunk 5 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: Just updated the docs using all of the suggestions above.

File: doc/classes/BakedLightmap.xml [hunk 6]
```diff
@@ -18,12 +18,14 @@
 			<argument index="1" name="create_visual_debug" type="bool" default="false">
 			</argument>
 			<description>
+				Bakes the lightmaps within the currently-editted scene.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]: Sorry to bother again, but there's still the duplicated "t" from before :slightly_smiling_face:

```suggestion
				Bakes the lightmaps within the currently-edited scene.
```

File: doc/classes/BakedLightmap.xml [hunk 7]
```diff
@@ -18,12 +18,14 @@
...
 			</description>
 		</method>
 		<method name="debug_bake">
 			<return type="void">
 			</return>
 			<description>
+				Executes a dry run bake of lightmaps within the currently-editted scene.
```
Comment (Calinou) [inline doc/classes/BakedLightmap.xml]: Same as above:

```suggestion
				Executes a dry run bake of lightmaps within the currently-edited scene.
```

**Review by Calinou — DISMISSED** [review]
Should be good to go after fixing the typos.

File: doc/classes/BakedLightmap.xml (hunk 7 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: Oops, good catch! Fixed :)

File: doc/classes/BakedLightmap.xml (hunk 6 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: Oops, good catch! This should be fixed now :)

File: doc/classes/BakedLightmap.xml [hunk 8]
```diff
@@ -32,8 +34,10 @@
 			Grid subdivision size for lightmapper calculation. The default value will work for most cases. Increase for better lighting on small details or if your scene is very large.
 		</member>
 		<member name="bake_default_texels_per_unit" type="float" setter="set_bake_default_texels_per_unit" getter="get_bake_default_texels_per_unit" default="20.0">
+			If a [code]lightmap_size_hint[/code] isn't specified on a [code]Mesh[/code], the lightmap baker will dynamically set the lightmap size using this value. This value is measured in texels per world unit. The maximum lightmap texture size is 4096x4096.
```
Comment (akien-mga) [inline doc/classes/BakedLightmap.xml]: You can use `[member Mesh.lightmap_size_hint]` to have a direct reference to that property's docs, and `[Mesh]` to link the Mesh docs.

File: doc/classes/BakedLightmap.xml [hunk 9]
```diff
@@ -18,12 +18,14 @@
 			<argument index="1" name="create_visual_debug" type="bool" default="false">
 			</argument>
 			<description>
+				Bakes the lightmaps within the currently-edited scene.
```
Comment (akien-mga) [inline doc/classes/BakedLightmap.xml]: There should not be an hyphen after an adverb ending in `-ly`, see https://www.dailywritingtips.com/adverbs-and-hyphens/
Same below.

File: doc/classes/BakedLightmap.xml (hunk 8 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: Thanks for the tip Remi, this should be updated now :)

File: doc/classes/BakedLightmap.xml (hunk 9 above)
Comment (NathanWarden) [reply doc/classes/BakedLightmap.xml]: I've removed the hyphen on both of these.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**